src/octoprint/static/js/app/viewmodels/connection.js

```diff
@@ -97,15 +106,10 @@ $(function() {
             self.isReady(data.flags.ready);
             self.isLoading(data.flags.loading);
 
-            var connectionTab = $("#connection");
-            if (self.previousIsOperational != self.isOperational()) {
-                if (self.isOperational() && connectionTab.hasClass("in")) {
-                    // connection just got established, close connection tab for now
-                    connectionTab.collapse("hide");
-                } else if (!connectionTab.hasClass("in")) {
```

Comment:
I also changed this line when I moved it. I think there's another bug here. At least, this isn't idempotent, because if the first isOperational is true but the tab has already been hidden, the else if doesn't check the isOperational again and will just show the tab incorrectly.